tracer/src/Datadog.Trace/Agent/DiscoveryService/DiscoveryService.cs

```diff
@@ -42,10 +44,18 @@ private DiscoveryService(IApiRequestFactory apiRequestFactory)
 
         public string AgentVersion { get; private set; }
 
-        public static DiscoveryService Create(IApiRequestFactory apiRequestFactory)
+        public static DiscoveryService Create(ImmutableExporterSettings exporterSettings)
         {
             lock (GlobalLock)
             {
+                var apiRequestFactory = AgentTransportStrategy.Get(
+                    exporterSettings,
+                    productName: "discovery",
+                    tcpTimeout: null,
```

Comment:
Definitely don't want `null` here, that's a 100s timeout 😄 @zacharycmontoya has a PR to set all the default timeouts to 15s I think, maybe go with that for now?

Reply:
I don't have a strong opinion here, changed to `15s` here 45eb61aa0314287522a7d1da6c1ab223df349342